prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -27,6 +32,50 @@ import kotlin.reflect.jvm.isAccessible
  * A collection of helper functions that modify an existing FHIR bundle.
  */
 const val conditionExtensionurl = "https://reportstream.cdc.gov/fhir/StructureDefinition/reportable-condition"
+const val conditionCodeExtensionURL = "https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code"
+
+private fun lookupCondition(code: Coding, metadata: Metadata): Coding? {
+    val mappingTable = metadata.findLookupTable("observation-mapping").also {
+        if (it == null) {
+            return null // TODO: better error if table is missing? log vs throw?
+        }
+    }!!
+    val condition = mappingTable.dataRowsMap.find {
+        it[ObservationMappingConstants.TEST_CODE_KEY.lowercase()] == code.code // TODO: why lower
+    }
+    return if (condition.isNullOrEmpty()) {
+        null
+    } else {
+        Coding(
+            condition[ObservationMappingConstants.CONDITION_CODE_SYSTEM_KEY.lowercase()], // TODO: why lower
+            condition[ObservationMappingConstants.CONDITION_CODE_KEY.lowercase()], // TODO: why lower
+            condition[ObservationMappingConstants.CONDITION_NAME_KEY.lowercase()] // TODO: why lower
+        )
+    }
+}
+
+fun Observation.addMappedCondition(metadata: Metadata): List<ActionLogDetail> {
+    val codeSourcesMap = mapOf(
+        ObservationMappingConstants.BUNDLE_CODE_IDENTIFIER to this.code.coding,
+        ObservationMappingConstants.BUNDLE_CODEABLE_IDENTIFIER to this.valueCodeableConcept.coding
+    ).filterValues { it.isNotEmpty() }
+    if (codeSourcesMap.values.flatten().isEmpty()) return listOf(UnmappableConditionMessage())
+
+    return codeSourcesMap.mapNotNull { codeSourceEntry ->
+        codeSourceEntry.value.mapNotNull { code ->
+            lookupCondition(code, metadata).let { conditionCode ->
```

Comment:
I don't think you need the `let` here?

Reply:
If the lookup results in null we want to pass on the code.
Otherwise we want to add the extension and pass on a null that will get filtered.

Is there another scope function I should be using instead?

Reply:
Got it...is there maybe a missing test then?  I dropped the let and I think they all passed